migrations/020.do.publish-rsr.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE unpublished_rsr_rounds (
+  round_index NUMERIC PRIMARY KEY,
```

Comment:
❗ 

I realised one important thing: the round index is local to the Meridian smart contract version. When we deploy a new Meridian version, round index resets back to zero.

I propose to add the smart contract address to the per-round data inside the evaluation commitment.

In other words, round id should be a tuple `(contractAddress, roundIndex)`.

This affects the committed JSON structure too: Instead of converting this tuple to a string id that can be used as a key (property name), I think it will be better to change “rounds” from an object to an array, as was proposed in the call yesterday (or on Slack?).

Notion discussion:
https://www.notion.so/Design-Spark-SLI-on-chain-interface-121837df73d4801aaaa4f13ff7839cd0?d=b58a1a22a5d248258cae0bbb5c4d9620&pvs=4#b70f9a9b99dd43739b136393d08c2091

Here is the schema we use in spark-api's table `spark_rounds` - see `meridian_address` and `meridian_round`.

```
spark=# \d+ spark_rounds;
                                                     Table "public.spark_rounds"
       Column       |           Type           | Collation | Nullable | Default 
--------------------+--------------------------+-----------+----------+---------
 id                 | bigint                   |           | not null |         
 created_at         | timestamp with time zone |           | not null |         
 meridian_address   | text                     |           |          |         
 meridian_round     | bigint                   |           |          |         
 max_tasks_per_node | integer                  |           | not null | 360     
 start_epoch        | bigint                   |           | not null |         
 measurement_count  | bigint                   |           |          |         
```

The column `id` contains the round number that we create internally inside spark-api by tracking smart contract versions and mapping per-contract-round-index to this round number. It seemed like a good idea when I started building Spark. I discovered the limitations of that approach very soon and switched to `(meridian_address, meridian_rond)` in most places.